src/cow-react/api/gnosisProtocol/gasFeeApi.ts

```diff
@@ -0,0 +1,120 @@
+import { DEFAULT_NETWORK_FOR_LISTS } from 'constants/lists'
+import { SupportedChainId as ChainId } from 'constants/chains'
+import { GAS_FEE_ENDPOINTS, GAS_API_KEYS } from 'constants/index'
+
+// Values are returned as floats in gwei
+const ONE_GWEI = 1_000_000_000
+
+export type GasFeeEndpointResponse = {
+  lastUpdate: string
+  average: number | null
+  fast: number | null
+  slow: number | null
+}
+
+export interface EstimatedPrice {
+  confidence: number
+  price: number
+  maxFeePerGas: number
+  maxPriorityFeePerGas: number
+}
+
+type GasPrices = Omit<GasFeeEndpointResponse, 'lastUpdate'>
+
+// Here the key is what we use in gas state and the value is confidence level
+// coming from the Blockscout api, so we map state values with confidence lvls
+const priceMap: GasPrices = {
+  fast: 99,
+  average: 90,
+  slow: 70,
+}
+
+class GasFeeApi {
+  getUrl(chainId: ChainId) {
+    const baseUrl = GAS_FEE_ENDPOINTS[chainId]
+
+    if (chainId !== ChainId.GNOSIS_CHAIN) {
+      return `${baseUrl}?chainId=${chainId}`
```

Comment:
Can we always add chainId to the URL?

Reply:
I don't think so, because for Gnosis Chain we use Blockscout which has different url https://blockscout.com/xdai/mainnet/api/v1/gas-price-oracle